libnmstate/nm/device.py

```diff
@@ -343,6 +347,52 @@ def _delete_connection_callback(src_object, result, user_data):
             'dev={}, error=unknown'.format(devname))
 
 
+def delete_device(nmdev):
+    mainloop = nmclient.mainloop()
+    devpath = nmdev.get_path()
+    mainloop.push_action(_safe_delete_device_async, devpath)
+
+
+def _safe_delete_device_async(devpath):
+    mainloop = nmclient.mainloop()
+    client = nmclient.client()
+    nmdev = client.get_device_by_path(devpath)
+    if not nmdev:
+        # No callback is expected, so we should call the next one.
```

Comment:
This commit irritates me. How about `Nothing to do since the device is already gone` - the callback is a technical detail and does not explain the actual idea.

Reply:
Done